js/ui/grid_core/ui.grid_core.columns_controller.js

```diff
@@ -2408,11 +2437,17 @@ module.exports = {
                  */
                 addColumn: function(options) {
                     var that = this,
-                        column = createColumn(that, options);
-
-                    column.added = options;
+                        column = createColumn(that, options),
+                        index = that._columns.length;
 
                     that._columns.push(column);
+
+                    if(column.isBand) {
+                        that._columns = createColumnsFromOptions(that, that._columns);
+                        column = that._columns[index];
+                    }
+
+                    setAddedOption(that, index, [options]);
```

Comment:
Need to write test with state storing and with added banded column. Maybe recursion it is not necessary.